File: goclient/goclient/metricsjson/metrics_json_request_builder.go
package metricsjson

import (
    "context"
    i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f "github.com/microsoft/kiota-abstractions-go"
    id6760cbde1c5ad1214c40ac9ca069164ef0f31f8ecb790d183bca1f74a71e629 "github.com/andrueastman/untypednodesamples/goclient/goclient/models"
)

// MetricsJsonRequestBuilder builds and executes requests for operations under \metrics.json
type MetricsJsonRequestBuilder struct {
    i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.BaseRequestBuilder
}
// MetricsJsonRequestBuilderGetRequestConfiguration configuration for the request such as headers, query parameters, and middleware options.
type MetricsJsonRequestBuilderGetRequestConfiguration struct {
    // Request headers
    Headers *i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.RequestHeaders
    // Request options
    Options []i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.RequestOption
}
// NewMetricsJsonRequestBuilderInternal instantiates a new MetricsJsonRequestBuilder and sets the default values.
func NewMetricsJsonRequestBuilderInternal(pathParameters map[string]string, requestAdapter i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.RequestAdapter)(*MetricsJsonRequestBuilder) {
    m := &MetricsJsonRequestBuilder{
        BaseRequestBuilder: *i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.NewBaseRequestBuilder(requestAdapter, "{+baseurl}/metrics.json", pathParameters),
    }
    return m
}
// NewMetricsJsonRequestBuilder instantiates a new MetricsJsonRequestBuilder and sets the default values.
func NewMetricsJsonRequestBuilder(rawUrl string, requestAdapter i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.RequestAdapter)(*MetricsJsonRequestBuilder) {
    urlParams := make(map[string]string)
    urlParams["request-raw-url"] = rawUrl
    return NewMetricsJsonRequestBuilderInternal(urlParams, requestAdapter)
}
// Get some basic metrics for the entire directory.Just stunning numbers to put on a front page and are intended purely for WoW effect :)
// returns a Metricsable when successful
func (m *MetricsJsonRequestBuilder) Get(ctx context.Context, requestConfiguration *MetricsJsonRequestBuilderGetRequestConfiguration)(id6760cbde1c5ad1214c40ac9ca069164ef0f31f8ecb790d183bca1f74a71e629.Metricsable, error) {
    requestInfo, err := m.ToGetRequestInformation(ctx, requestConfiguration);
    if err != nil {
        return nil, err
    }
    res, err := m.BaseRequestBuilder.RequestAdapter.Send(ctx, requestInfo, id6760cbde1c5ad1214c40ac9ca069164ef0f31f8ecb790d183bca1f74a71e629.CreateMetricsFromDiscriminatorValue, nil)
    if err != nil {
        return nil, err
    }
    if res == nil {
        return nil, nil
    }
    return res.(id6760cbde1c5ad1214c40ac9ca069164ef0f31f8ecb790d183bca1f74a71e629.Metricsable), nil
}
// ToGetRequestInformation some basic metrics for the entire directory.Just stunning numbers to put on a front page and are intended purely for WoW effect :)
// returns a *RequestInformation when successful
func (m *MetricsJsonRequestBuilder) ToGetRequestInformation(ctx context.Context, requestConfiguration *MetricsJsonRequestBuilderGetRequestConfiguration)(*i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.RequestInformation, error) {
    requestInfo := i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.NewRequestInformationWithMethodAndUrlTemplateAndPathParameters(i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.GET, m.BaseRequestBuilder.UrlTemplate, m.BaseRequestBuilder.PathParameters)
    if requestConfiguration != nil {
        requestInfo.Headers.AddAll(requestConfiguration.Headers)
        requestInfo.AddRequestOptions(requestConfiguration.Options)
    }
    requestInfo.Headers.TryAdd("Accept", "application/json")
    return requestInfo, nil
}
// WithUrl returns a request builder with the provided arbitrary URL. Using this method means any other path or query parameters are ignored.
// returns a *MetricsJsonRequestBuilder when successful
func (m *MetricsJsonRequestBuilder) WithUrl(rawUrl string)(*MetricsJsonRequestBuilder) {
    return NewMetricsJsonRequestBuilder(rawUrl, m.BaseRequestBuilder.RequestAdapter);
}

File: javaclient/app/src/main/java/javaclient/app/metricsjson/MetricsJsonRequestBuilder.java
package javaclient.client.metricsjson;

import com.microsoft.kiota.BaseRequestBuilder;
import com.microsoft.kiota.BaseRequestConfiguration;
import com.microsoft.kiota.HttpMethod;
import com.microsoft.kiota.RequestAdapter;
import com.microsoft.kiota.RequestInformation;
import com.microsoft.kiota.RequestOption;
import com.microsoft.kiota.serialization.Parsable;
import com.microsoft.kiota.serialization.ParsableFactory;
import java.util.Collection;
import java.util.HashMap;
import java.util.Map;
import java.util.Objects;
import javaclient.client.models.Metrics;
/**
 * Builds and executes requests for operations under /metrics.json
 */
@jakarta.annotation.Generated("com.microsoft.kiota")
public class MetricsJsonRequestBuilder extends BaseRequestBuilder {
    /**
     * Instantiates a new {@link MetricsJsonRequestBuilder} and sets the default values.
     * @param pathParameters Path parameters for the request
     * @param requestAdapter The request adapter to use to execute the requests.
     */
    public MetricsJsonRequestBuilder(@jakarta.annotation.Nonnull final HashMap<String, Object> pathParameters, @jakarta.annotation.Nonnull final RequestAdapter requestAdapter) {
        super(requestAdapter, "{+baseurl}/metrics.json", pathParameters);
    }
    /**
     * Instantiates a new {@link MetricsJsonRequestBuilder} and sets the default values.
     * @param rawUrl The raw URL to use for the request builder.
     * @param requestAdapter The request adapter to use to execute the requests.
     */
    public MetricsJsonRequestBuilder(@jakarta.annotation.Nonnull final String rawUrl, @jakarta.annotation.Nonnull final RequestAdapter requestAdapter) {
        super(requestAdapter, "{+baseurl}/metrics.json", rawUrl);
    }
    /**
     * Some basic metrics for the entire directory.Just stunning numbers to put on a front page and are intended purely for WoW effect :)
     * @return a {@link Metrics}
     */
    @jakarta.annotation.Nullable
    public Metrics get() {
        return get(null);
    }
    /**
     * Some basic metrics for the entire directory.Just stunning numbers to put on a front page and are intended purely for WoW effect :)
     * @param requestConfiguration Configuration for the request such as headers, query parameters, and middleware options.
     * @return a {@link Metrics}
     */
    @jakarta.annotation.Nullable
    public Metrics get(@jakarta.annotation.Nullable final java.util.function.Consumer<GetRequestConfiguration> requestConfiguration) {
        final RequestInformation requestInfo = toGetRequestInformation(requestConfiguration);
        return this.requestAdapter.send(requestInfo, null, Metrics::createFromDiscriminatorValue);
    }
    /**
     * Some basic metrics for the entire directory.Just stunning numbers to put on a front page and are intended purely for WoW effect :)
     * @return a {@link RequestInformation}
     */
    @jakarta.annotation.Nonnull
    public RequestInformation toGetRequestInformation() {
        return toGetRequestInformation(null);
    }
    /**
     * Some basic metrics for the entire directory.Just stunning numbers to put on a front page and are intended purely for WoW effect :)
     * @param requestConfiguration Configuration for the request such as headers, query parameters, and middleware options.
     * @return a {@link RequestInformation}
     */
    @jakarta.annotation.Nonnull
    public RequestInformation toGetRequestInformation(@jakarta.annotation.Nullable final java.util.function.Consumer<GetRequestConfiguration> requestConfiguration) {
        final RequestInformation requestInfo = new RequestInformation(HttpMethod.GET, urlTemplate, pathParameters);
        requestInfo.configure(requestConfiguration, GetRequestConfiguration::new);
        requestInfo.headers.tryAdd("Accept", "application/json");
        return requestInfo;
    }
    /**
     * Returns a request builder with the provided arbitrary URL. Using this method means any other path or query parameters are ignored.
     * @param rawUrl The raw URL to use for the request builder.
     * @return a {@link MetricsJsonRequestBuilder}
     */
    @jakarta.annotation.Nonnull
    public MetricsJsonRequestBuilder withUrl(@jakarta.annotation.Nonnull final String rawUrl) {
        Objects.requireNonNull(rawUrl);
        return new MetricsJsonRequestBuilder(rawUrl, requestAdapter);
    }
    /**
     * Configuration for the request such as headers, query parameters, and middleware options.
     */
    @jakarta.annotation.Generated("com.microsoft.kiota")
    public class GetRequestConfiguration extends BaseRequestConfiguration {
    }
}
